docs/knowledge_base/tutorials/version_management.md

```diff
@@ -0,0 +1,104 @@
+When developing and deploying a Taipy application, it is straightforward to manage Taipy entities
+(scenarios, tasks, data nodes, etc.) and keeps them up-to-date when the configuration changes.
+
+In the following, we will use a basic Taipy application example defined in `main.py`.
+
+```python linenums="1" title="main.py"
+{%
+include-markdown "../../manuals/core/versioning/code_example/main.py"
+comments=false
+%}
+```
+
+Prerequisites of git is required to follow this tutorial.
+
+# Initialize a Taipy application
+
+Inside your application directory, let's git initialize the application directory by running `git init`:
+
+```console
+$ git init
+...
+Initialized empty Git repository in ~/your_taipy_application/.git/
+```
+
+We then need to create a `.gitignore` file to ignore the `.data` directory that contains Taipy
+entities when committing the application to Git. You can create the `.gitignore` file manually
+or by running the following command:
+
+```console
+$ echo ".data" > .gitignore
+```
+Then we can commit the `.gitignore` file to Git:
+
+```console
+$ git add .gitignore
+$ git commit -m "initialize .gitignore that ignore taipy entities"
+```
+
+Now you're ready to manage your Taipy application with Git and Taipy version management.
+
+# Create a Taipy application version
+
+By default, a Taipy application runs in development mode, which means entities from previous
+development run are deleted before running the application. To save the entities of a run, you
+can create a new experiment version of your application by running your Taipy application with
+`--experiment` option on the CLI. Let's name the experiment version `1.0` and commit the version
+to Git.
+
+```console
+$ taipy run main.py --experiment 1.0
+$ git add main.py
+$ git commit -m "create experiment version 1.0"
+```
+
+# Switching between versions
+
+A commonly used git workflow is to use `git checkout` to switch to a different branch and work on
+a new application version. Let's create a new git branch `experiment-1.1` and switch to it:
+
+```console
+$ git checkout -b experiment-1.1
+Switched to a new branch 'experiment-1.1'
+```
```

Comment:
Actually, since you git ignore the `.data` folder, there is nothing related to exepriment in the committed code. I would simply call the branch `1.1`

Reply:
I'm not sure to understand what you mean

Reply:
I mean it is strange for me to name the branch with the prefix experiment. Why not just call it `1.1`. The code itself (what is contained in the git branch) has nothing related to the fact it has run in experiment mode at first. For instance, you can run the code in a development or production mode.

Reply:
Ah, that makes sense 😅 Will do that